mlflow/models/model.py

```diff
@@ -1176,7 +1179,7 @@ def log_model_metrics_for_step(client, model_id, run_id, step):
                     tags={key: str(value) for key, value in tags.items()}
                     if tags is not None
                     else None,
-                    flavor=flavor.__name__ if hasattr(flavor, "__name__") else "custom",
+                    flavor=flavor_name,
```

Comment:
I wondered if there any side effect when we change "pyfunc" to more granular naming in `_create_logged_model`, but it seems this was added only for the logging purpose